src/main/java/frc/utils/time/TimeUtils.java

```diff
@@ -0,0 +1,47 @@
+package frc.utils.time;
+
+import edu.wpi.first.hal.HALUtil;
+import frc.utils.Conversions;
+import frc.utils.alerts.Alert;
+import frc.utils.alerts.AlertManager;
+import frc.utils.alerts.PeriodicAlert;
+import org.littletonrobotics.junction.Logger;
+
+public class TimeUtils {
+
+	public static final double DEFAULT_CYCLE_TIME_SECONDS = 0.02;
+
+	private static double lastCycleTimeSeconds = 0;
+	private static double newCycleTimeSeconds = 0;
+
+	static {
+		AlertManager.addAlert(
```

Comment:
Consider adding an alert for timestamps that are too old (3s>) or that are newer than the current time. This could catch errors in the timestamp's calculation.

Reply:
Where?

Reply:
nvm